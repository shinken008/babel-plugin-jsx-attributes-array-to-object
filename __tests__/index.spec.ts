import * as babel from '@babel/core'
import syntaxJSX from '@babel/plugin-syntax-jsx'
import plugin from '../src'

const plugins = [
  syntaxJSX,
  [plugin, { attributes: ['style'] }],
]

it('should work!', function () {
  const code = `
  var a = { color: 'red' };
  
  <div style={[a, { color: 'gray' }]}></div>
  `
  const result = babel.transform(code, { plugins })
  expect(result?.code).toMatchSnapshot()
})
