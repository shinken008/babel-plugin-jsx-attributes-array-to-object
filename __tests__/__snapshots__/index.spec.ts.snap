// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should work! 1`] = `
"var a = {
  color: 'red'
};
<div style={__mergeObject(a, {
  color: 'gray'
})}></div>;

function __mergeObject(...args) {
  return args.reduce((obj, next) => {
    obj = Object.assign(Object.assign({}, obj), next);
    return obj;
  }, {});
}"
`;
